src/adapter/base.cr

```diff
@@ -8,7 +8,7 @@ abstract class Granite::Adapter::Base
   property database : DB::Database
 
   def initialize(adapter : String)
-    if url = ENV["DATABASE_URL"]? || replace_env_vars(settings(adapter)["database"].to_s)
+    if url = ENV["DATABASE_URL"]? || Granite::ORM.settings.database_url || replace_env_vars(settings(adapter)["database"].to_s)
```

Comment:
I would consider removing the `ENV["DATABASE_URL"]`, Reasons:

- Any system using granite can explicitly set the URL via `Granite::ORM.settings.database_url` including `Granite::ORM.settings.database_url = ENV["DATABASE_URL"]`
- If a DATABASE_URL evn variable is define in the system that is should NOT be use by granite it will fail to load the db url set via `Granite::ORM.settings.database_url`. Think multi-tenant apps where there is a DB holding config settings that is shared across apps and then each app has its own individual db.
- With ENV["DATABASE_URL"] Granite knows to much about `where` to look and violates `tell don't ask` principle. Now that we can set database url by composition with `Granite::ORM.settings.database_url` we can tell granite which connection to use. 

`ENV["DATABASE_URL"]` IMHO should be remove.

Reply:
@eliasjpr - So you are saying that amber (or another app) should make use of `ENV["DATABASE_URL"]`? Rather than directly in granite?

I could see that. If an app wanted to start with `ENV["DATABASE_URL"]` and append options to it, then Granite would be overriding those options.

```
# In amber (or other app)
Granite::ORM.settings.database_url = ENV["DATABASE_URL"] + "?max_connections=#{ENV["CPU_COUNT"]}"
```

Reply:
In terms of amber I think that makes sense, but if people will use granite other places then it might be easier to leave it how it is.

Reply:
What about changing the priority?
```
Granite::ORM.settings.database_url || ENV["DATABASE_URL"]? || replace_env_vars(settings(adapter)["database"].to_s)
```

Reply:
It feels like magic that I set an environment variable and Granite just connects to it, unless someone knows this about granite it would be hard to figure it out. With the existence of `Granite::ORM.settings.database_url` I don't see the need of having `ENV["DATABASE_URL"]` anymore, that's all. 

As per @elorest comment if people use granite in other places is the same input they would need `Granite::ORM.settings.database_url = ENV["DATABASE_URL"]` instead of having Granite connecting implicitly to a database.

This is what I think and most likely I could be wrong, it is up to you guys if you think is worth the change.

Reply:
Amethyst-Model, Kemalyst-Model, Granite::ORM is following the Rails database configuration.  http://edgeguides.rubyonrails.org/configuring.html#configuring-a-database

Rails developers are familiar with these locations for configuring their database.  Heroku also uses the `DATABASE_URL` environment variable for easy deployment.  I would prefer keeping these for people that prefer the familiarity of Rails (i.e. me)

We can look at deprecating `config/database.yml` but I would still want to do this in a progressive manner.  I also want to keep the DATABASE_URL environment variable as the override since this is fairly well known and commonly used (by me).

Again, Granite::ORM is being used outside of Amber including old Kemalyst sites.  Let's be kind to these people(me).  If you have a really compelling reason we should create a breaking change, I would be open to it but I don't see a good enough reason at this point.

Reply:
@drujensen would you consider this breaking?

```
if url = Granite::ORM.settings.database_url || ENV["DATABASE_URL"]? || ...
```

Reply:
No, probably wouldn't break existing apps.  But I would consider it incorrect since the expected behaviour is DATABASE_URL will override your config.  Anyway, that is the way it is in Rails.

Reply:
Does it mean that this won't be possible?
```
# In amber (or other app)
Granite::ORM.settings.database_url = ENV["DATABASE_URL"] + "?max_connections=#{ENV["CPU_COUNT"]}"
```
If so, is there an easy way to set the connection settings?
https://crystal-lang.org/docs/database/connection_pool.html#configuration